README.md

```diff
@@ -1,32 +1,504 @@
+
+```
+[[Swrve sharedInstance] currencyGiven:@"coins" givenAmount:25];
+```
+
+To ensure virtual currency events are not ignored by the server, make sure the currency name configured in your app matches exactly the Currency Name you enter in the App Currencies section on the App Settings screen (including case-sensitive). If there is any difference, or if you haven’t added the currency in Swrve, the event will be ignored and return an error event called Swrve.error.invalid_currency. Additionally, the ignored events will not be included in your KPI reports. For more information, see [Add Your App](http://docs.swrve.com/getting-started/add-your-app/).
+
+### Sending IAP Events and IAP Validation ###
+
+You must notify Swrve when an in-app purchase occurs as follows:
+
+```
+- (void) onTransactionProcessed:(SKPaymentTransaction*) transaction productBought:(SKProduct*) product {
+   // Tell Swrve what was purchased.
+   SwrveIAPRewards* rewards = [[SwrveIAPRewards alloc] init];
+ 
+   // You can track the purchase of virtual items
+   [rewards addItem:@"com.myapp.item_sku" withQuantity:1];
+   // Or virtual currency or both.
+   [rewards addCurrency:@"coins" withAmount:100];
+ 
+   // Send the IAP event to Swrve.
+   [[Swrve sharedInstance] iap:transaction product:product rewards:rewards];
+}
+```
+
+### Enabling IAP Receipt Validation
+
+Swrve automatically validates iOS in-app purchases and ignores any events that are considered fraudulent as per [Apple guidelines](https://developer.apple.com/library/ios/releasenotes/General/ValidateAppStoreReceipt/Introduction.html).
+
+To enable validation of IAP receipts against your app:
+
+1. On the Setup menu, click Integration Settings.
+
+2. Under IAP Validation, in the iTunes App Bundle Identifier section, enter your Apple Bundle Id.
+
+3. Click Save.
+
+Although Swrve can validate IAP receipts without this identifier, providing it enables Swrve to verify a receipt against the app itself for greater precision during validation. Swrve checks that the bundle ID included in an in-app purchase receipt corresponds to your bundle ID before calculating your revenue KPIs. This ensures that your revenue figures are as accurate as possible (ignoring pirates and cheaters).
+
+Generally, you enter the Apple bundle ID when configuring the Integration Settings screen as part of the Swrve onboarding process. You can edit the settings on this screen at any time.
+
+To access your bundle ID, access the information property list file (`<App>-Info.plist`) in your Xcode project.
+
+### Verifying IAP Receipt Validation
+
+Use the following events to monitor the success of IAP receipt validation:
+
+* `swrve.valid_iap` – fired if receipt verification has been successful and the receipt is valid.
+* `swrve.invalid_iap` – fired if receipt verification has been successful and the receipt is invalid.
+
+### Unvalidated IAP
```

Comment:
worth mentioning some use cases for this, eg. any purchases made from the app not via apple.  need to mention that any validation should be done before this is called, as any revenue sent by this event will automatically count